watertap/examples/chemistry/chem_scaling_utils.py

```diff
@@ -0,0 +1,119 @@
+###############################################################################
+# WaterTAP Copyright (c) 2021, The Regents of the University of California,
+# through Lawrence Berkeley National Laboratory, Oak Ridge National
+# Laboratory, National Renewable Energy Laboratory, and National Energy
+# Technology Laboratory (subject to receipt of any required approvals from
+# the U.S. Dept. of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/watertap/"
+#
+###############################################################################
+
+"""
+    This is a set of scaling utility helper functions used by the chemistry tests.
+"""
+from idaes.core.util import scaling as iscale
+from pyomo.environ import value
+
+__author__ = "Austin Ladshaw"
+
+## Helper function for setting eps values associated with solubility_product functions
+# NOTE: Function does nothing if no solubility reactions are present
+def _set_eps_vals(rxn_params, rxn_config, factor=1e-2):
+    # For solubility reactions, have to set the eps value
+    if (hasattr(rxn_params,"equilibrium_reaction_idx")):
+        for rid in rxn_params.equilibrium_reaction_idx:
+            # Grab the 'k_eq_ref' value from the reaction config
+            scale = rxn_config["equilibrium_reactions"][rid]["parameter_data"]["k_eq_ref"][0]
+
+            # NOTE: ONLY certain functions have an eps value that we need to set
+            if (hasattr(rxn_params.component("reaction_"+rid),"eps")):
+                # highest allowable value for setting eps based on k_eq_ref
+                if scale < 1e-16:
+                    rxn_params.component("reaction_"+rid).eps.value = scale*factor
+                else:
+                    rxn_params.component("reaction_"+rid).eps.value = 1e-16*factor
```

Comment:
Should `1e-16` also be settable from the function?
```suggestion
                # highest allowable value for setting eps based on k_eq_ref
                rxn_params.component("reaction_"+rid).eps.value = min(1e-16, scale)*factor
```

Reply:
I guess we 'could' let the 1e-16 be settable. I will consider changing that.